pkg/apiserver/certificate/certificate.go

```diff
@@ -110,5 +103,53 @@ func ApplyServerCert(selfSignedCert bool, client kubernetes.Interface, aggregato
 	}
 
 	caCertController := newCACertController(caContentProvider, client, aggregatorClient)
+
+	if selfSignedCert {
+		go RotateSelfSignedCertificates(context.Background(), caCertController, secureServing, rotateInterval)
```

Comment:
you can use `time.AfterFunc` here, and use it to call `RotateSelfSignedCertificates` after the desired time, in its own go routine.